shared/src/main/scala/org/atnos/eff/Eff.scala

```diff
@@ -296,12 +296,10 @@ trait EffInterpretation {
       case Pure(a, Last(None))    => Monad[M].pure(Right(a))
 
       case Impure(u, continuation, last) =>
-        u match {
-          case Union1(ta) =>
-            last match {
-              case Last(Some(l)) => ta.map(x => Left(continuation(x).addLast(last)))
-              case Last(None)    => ta.map(x => Left(continuation(x)))
-            }
+        val ta = u.tagged.valueUnsafe.asInstanceOf[M[A]]
```

Comment:
`.tagged` is an unsafe operation. At this stage how do you know that you have a `UnionTagged` and not a `UnionAppend`?